scripts/evaluate_chunk_size.py

```diff
@@ -0,0 +1,148 @@
+import os
+import uuid
+import time
+
+import chromadb
+from openai import OpenAI
+
+
+EMBEDDING_COLLECTION_NAME = "learn"
+chunk_sizes = ["paragraph", "half_a_paragraph", "quarter_a_paragraph", "sentence"]
+
+chromadb_client = chromadb.HttpClient(
+    host=os.getenv("CHROMADB_HOST", "chromadb"),
+    port=8000,
+    ssl=False,
+    headers=None,
+    settings=chromadb.config.Settings(),
+    tenant=chromadb.config.DEFAULT_TENANT,
+    database=chromadb.config.DEFAULT_DATABASE,
+)
+
+openai_client = OpenAI(api_key=os.getenv("OPENAI_API_KEY"))
+chromadb_client = chromadb.HttpClient(
+    host=os.getenv("CHROMADB_HOST", "chromadb"),
+    port=8000,
+    ssl=False,
+    headers=None,
+    settings=chromadb.config.Settings(),
+    tenant=chromadb.config.DEFAULT_TENANT,
+    database=chromadb.config.DEFAULT_DATABASE,
+)
+
+# Questions to ask about the text
+f_query = open("../data/RAG_chunk_evaluation_questions.txt", "r")
+my_queries = f_query.read().split("\n")
+f_query.close()
+
+unique_file_id = str(int(time.time()))
+
+for chunk_size_desc in chunk_sizes:
+
+    print(chunk_size_desc)
+
+    # a file to store responses
+    save_path = "evaluate_chunk_size_results"
+    if not os.path.exists(save_path):
+        os.makedirs(save_path)
+    f_response = open(
+        save_path + "/response_" + unique_file_id + "_" + chunk_size_desc + ".txt", "w"
+    )
+    f_response.writelines([chunk_size_desc, "\n"])
+
+    # read and chunk the text
+    f = open("../data/chapterXX.txt", "r")
+    chunk_sizes = ["paragraph", "half_a_paragraph", "quarter_a_paragraph", "sentence"]
+    if chunk_size_desc == "paragraph":
+        my_documents = f.read().split("\n")
+        my_ids = [f"{uuid.uuid4().hex}" for _ in range(len(my_documents))]
+    elif chunk_size_desc == "half_a_paragraph":
+        my_documents_tmp = f.read().split("\n")
+        my_documents = []
+        for i in range(
+            len(my_documents_tmp)
+        ):  # Computationally heavy. Let's look for a python library.
+            for c in range(2):
+                half = len(my_documents_tmp[i]) // 2
+                my_documents.append(my_documents_tmp[i][c * half : (c + 1) * half])
+        my_ids = [f"{uuid.uuid4().hex}" for _ in range(len(my_documents))]
+    elif chunk_size_desc == "quarter_a_paragraph":
+        my_documents_tmp = f.read().split("\n")
+        my_documents = []
+        for i in range(
+            len(my_documents_tmp)
+        ):  # Computationally heavy. Let's look for a python library.
+            for c in range(4):
```

Comment:
`nit` `c` looks like a character (`str` with length = 1) because it is a common pattern to use `c` to iterate through a string.

```python
for c in "test string":
  print(c)
```

I'm bad at naming, too, but I would use `count`, `j`, or `quarter_index` if I were to write this script